src/peft/tuners/lora/model.py

```diff
@@ -126,32 +125,7 @@ def _check_new_adapter_config(self, config: LoraConfig) -> None:
 
     @staticmethod
     def _check_target_module_exists(lora_config, key):
-        if isinstance(lora_config.target_modules, str):
-            target_module_found = re.fullmatch(lora_config.target_modules, key)
-        else:
-            target_module_found = any(
-                re.match(f".*\.{target_key}$", key) for target_key in lora_config.target_modules
-            ) or any(target_key == key for target_key in lora_config.target_modules)
-            is_using_layer_indexes = getattr(lora_config, "layers_to_transform", None) is not None
-            layer_indexing_pattern = getattr(lora_config, "layers_pattern", None)
-
-            if is_using_layer_indexes and target_module_found:
-                layers_pattern = COMMON_LAYERS_PATTERN if layer_indexing_pattern is None else layer_indexing_pattern
-                layers_pattern = [layers_pattern] if isinstance(layers_pattern, str) else layers_pattern
-
-                for pattern in layers_pattern:
-                    layer_index = re.match(f".*.{pattern}\.(\d+)\.*", key)
-                    if layer_index is not None:
-                        layer_index = int(layer_index.group(1))
-                        if isinstance(lora_config.layers_to_transform, int):
-                            target_module_found = layer_index == lora_config.layers_to_transform
-                        else:
-                            target_module_found = layer_index in lora_config.layers_to_transform
-
-                        break
-                    else:
-                        target_module_found = False
-        return target_module_found
+        return check_target_module_exists(lora_config, key)
```

Comment:
we could replace this static method to a call to the public method - we can do that in. a follow up PR - wdyt @BenjaminBossan @pacman100 ?

Reply:
I think I'd leave it as is for now. We may still want to extend the low level API with regard to adding custom tuners/adapter layers, and having public methods that relate to that area increases the difficulty of making these changes without breaking backwards compatibility.